src/App.tsx

```
@@ -1,24 +1,169 @@
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [hasErrors, setHasErrors] = useState(false);
+  const [error, setError] = useState<ErrorSpec | null>(null);
+  const [status, setStatus] = useState<Status>(Status.ALL);
+  const [uncompletedTodosCount, setUncompletedTodosCount] = useState<number>(0);
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState<Todo | null>(null);
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todosFromServer, status),
+    [status, todosFromServer],
+  );
+
+  useEffect(() => {
+    client
+      .get<Todo[]>(ADDED_URL)
+      .then((todos) => {
+        setTodosFromServer(todos);
+
+        if (todos.length === 0) {
+          setError(ErrorSpec.EMPTY_TITLE);
+        }
+      })
+      .catch(() => setError(ErrorSpec.NOT_LOADED));
+  }, []);
+
+  useEffect(() => {
+    const completedCount = todosFromServer.filter(
+      (todo) => !todo.completed,
+    ).length;
+
+    setUncompletedTodosCount(completedCount);
+  }, [todosFromServer, status, filteredTodos]);
+
+  const handleStatus = (newStatus: Status) => {
+    setStatus(newStatus);
+  };
+
+  const cleanErrors = () => {
+    setHasErrors(false);
+  };
+
+  const addTodo = (title: string) => {
+    setInputDisabled(true);
+    const preparedTitle = title.trim();
+
+    if (!preparedTitle) {
+      setError(ErrorSpec.EMPTY_TITLE);
+      setHasErrors(true);
+      setInputDisabled(false);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    };
+
+    TodoService.createTodo(newTodo)
+      .then((responce) => {
+        setTempTodo(responce);
+        setTodosFromServer([...todosFromServer, responce]);
+      })
+      .catch(() => {
+        setError(ErrorSpec.NOT_ADDED);
+        setInputDisabled(false);
+        throw new Error();
```

You already set the error text above, if you also want to throw the error into the console - you need to pass the `error` object from `catch` or duplicate the text, leaving the `Error` blank is definitely not an option.
Fix it in all code.